File: leetcode/624.maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        // arrays already sorted
        int ans = 0;
        int a = -1000000;
        int b = 1000000;
        for (const vector<int>& arr: arrays) {
            ans = max({ans, arr.back() - b, a - arr.front()});
            b = min(b, arr.front());
            a = max(a, arr.back());
        }
        return ans;
    }
};

File: leetcode/go/624.maximum-distance-in-arrays.go
func maxDistance(arrays [][]int) int {
    ans, mini, maxi, last := 0, 1000000, -1000000, 0
    for _, arr := range arrays {
        last = len(arr)-1
        ans = max(ans, maxi - arr[0], arr[last] - mini)
        maxi = max(arr[last], maxi)
        mini = min(arr[0], mini)
    }
    return ans
}

